### PR TITLE
rpc for worker callback phase 1

### PR DESCRIPTION
**Pull request by goodwanghan**

### CODE REVIEW

**Comment by goodwanghan** [comment]
When a transformer is working, we need to be able to collect information in realtime. For example, when do distributed tuning, we need to get model performance in realtime to decide if we want to kill that training, and the metrics can also be visualized on local driver

File: fugue_test/execution_suite.py [hunk 1]
```diff
@@ -553,6 +553,7 @@ def test_limit(self):
             d = e.limit(a, n=1, presort="a asc, b desc", partition_spec=ps)
             f = e.limit(a, n=1, presort=None, partition_spec=ps2)
             g = e.limit(a, n=2, presort="a desc", na_position="last")
+            h = e.limit(a, n=2, presort="a", na_position="first")
```
Comment (kvnkho) [inline fugue_test/execution_suite.py:556]: This test is pretty similar to c. Are you test if presort works with just a letter?

Comment (goodwanghan) [reply fugue_test/execution_suite.py:556]: no, this is to cover all code you wrote, you didn't cover this case, so the test coverage dropped to 99%

File: fugue_test/builtin_suite.py [hunk 2]
```diff
@@ -1176,6 +1176,37 @@ def o4(dfs: DataFrames) -> None:
...
+                    return str(self.n)
+
+            cb = Callbacks()
+            funcs = {"c": cb.call}
+
+            class CallbackTransformer(Transformer):
+                def get_output_schema(self, df):
+                    return df.schema
+
+                def transform(self, df):
```
Comment (kvnkho) [inline fugue_test/builtin_suite.py:1196]: nice!

**Review by kvnkho — APPROVED** [review]
Nice work! The set-up for Callbacks is very general too.

**Pull request merged**